jasmine/src/inverted-index.js

```diff
@@ -0,0 +1,149 @@
+class Index {
+
+    // Method to create an inverted-index of the documents
+    createIndex(filePath) {
+
+        // Here, self refers to the Index object
+        var self = this;
+
+        // Initialise Object to store the inverted-index
+        this.indexObj = {};
+
+        /* Fetch the document 
+         * Resolve the promise and store the data in a variable, 'finallySomeData'
+         */
+        return fetch(filePath).then(function(response) {
+            return response.json();
+
+        })
+
+        .then(function(finallySomeData) {
+            self.books = finallySomeData;
+
+            // Call the populateIndex function to create the inverted-index
+            populateIndex();
+
+        });
+
+        function populateIndex() {
+            // Iterate through the data obtaining each book's position in the document.
+            self.books.forEach((book, docIndex) => {
+
+                /* For each document, turn to string, lowercase, remove special characters
+                 * and trim beginning of line spaces.
+                 */
+                var bookObjectString = JSON.stringify(book)
+                    .toLowerCase().replace(/\W/g, ' ')
+                    .replace(/\s+/g, ' ').trim();
```

Comment:
- Stop words intentionally left in to  avoid a loss of context or meaning when searching for phrases.
- Test JSON intentionally used to run tests to ensure that the tests pass for all JSON files regardless of structure.
- Keys : 'title' and 'text' intentionally left in to avoid loss of meaning when the structure of the object changes. See additional object in books.JSON